spk/syncthing/src/syncthing.sc

```diff
@@ -1,4 +1,4 @@
-[syncthing]
+[syncthing-bep]
```

Comment:
Does this work? Did you try to uninstall syncthing and the port reservation was removed?

Regarding the dev guide this must be `syncthing_bep`
_If your package needs more than one port description, you can define service_name2, service_name3, …_

Reply:
There is a contradiction in the Developer Guide itself.  All examples use the underscore (`_`), whereas the description only allows `“a~z”, “A~Z”, “0~9”, “-”, “\”, “.”` (no underscore, only dash).

Reply:
The examples in the dev guide are correct. Only the service name does not allow `_`.

In the first example `[ss_findhostd_port]` is the service name `ss`
in the second example `[dhcp_udp]` is the service name `dhcp`.

Reply:
> Does this work? Did you try to uninstall syncthing and the port reservation was removed?

I just tried it on DSM 6.  During a package upgrade, the firewall rule stays unmodified and still displays the right description after the `[syncthing]` to `[syncthing-bep]` rename.  Subsequently uninstalling the package (without touching firewall rules) does remove the entry automatically.

Reply:
> The examples in the dev guide are correct. Only the service name does not allow `_`.

Sorry I don't understand at all.  AFAIU the whole "INI section" name within brackets is the service name.  I don't read anything in there about the first part before an underscore being somehow special.  It just says:

    service_name cannot be empty and can only include characters “a~z”, “A~Z”, “0~9”, “-”, “\”, “.”

`service_name` is the placeholder / "variable name", in our case `syncthing-bep`.  The value does not allow underscores, but of course they can use whatever placeholder string in the docs.  However the examples contradict their own rules.

I've played around with this a bit while testing the certificate implementation.  Certainly using a dash was not worse than an underscore, but I can't remember if there were any problems with using an underscore.  In the end I decided to take the value description as reference, disregarding their examples.

Reply:
IMHO the service name here is `syncthing` and not `syncthing-bep` as defined in `SERVICE_COMMAND`.

If you need multiple sections in *.sc files you can use servicename + `_` + postfix to create different sections with unique names. And this is optional if you want different descriptions for different ports of your service.

Sorry for not posting the first part of the dev description before:
```
Usually a package only has one unique service
name. If your package needs more than one port
description, you can define service_name2,
service_name3, …
```

It is not easy to write good documentation and synology badly uses `service_name` for the section.
It would make more sense with a better naming like `[service_postfix]` or `[service-name_postfix]`.

Summary:
The current syncthing package has the sections `[syncthing]`, `[syncthing-discovery]` and `[syncthing-webui]`.
This should be adjusted to `[syncthing_bep]`, `[syncthing_discovery]` and `[syncthing_webui]`.

Reply:
I still think using the underscore directly contradicts the rules given in the Developer Guide.  Apparently I'm not alone either, the `shairport-sync` spksrc package also uses a dash here.  While `rabbitmq` and `ffsync` use the underscore.

Reply:
@hgy59 I'm still leaning toward the version with dashes, the more often I re-read Synology's docs on the matter.  Can you please reconsider my argumentation?

There are some other unresolved review comments.  Could you please look at those again as well?